sipa/base.py

```diff
@@ -41,11 +41,11 @@ def babel_selector():
 
     if 'locale' in request.args and Locale(
             request.args['locale']) in possible_locales():
-        session['locale'] = request.args['locale']
-    elif not session.get('locale'):
+        return request.args['locale']
+    elif request.cookies.get("lang") is None:
         langs = []
         for lang in possible_locales():
             langs.append(lang.language)
-        session['locale'] = request.accept_languages.best_match(langs)
+            return request.accept_languages.best_match(langs)
 
-    return session.get('locale')
+    return request.cookies.get("lang", "en")
```

Comment:
You should check that `request.cookies['lang']` is in `possible_locales()`, like the if-branch does.

It would probably be best if you would structure the elif-branch like the if-branch and do the `for lang in possible_locales()` in an `else`branch.